doc/languages-frameworks/gnome.section.md

```diff
@@ -62,6 +62,18 @@ In the rare case you need to use icons from dependencies (e.g. when an app force
 
 To avoid costly file system access when locating icons, GTK, [as well as Qt](https://woboq.com/blog/qicon-reads-gtk-icon-cache-in-qt57.html), can rely on `icon-theme.cache` files from the themes' top-level directories. These files are generated using `gtk-update-icon-cache`, which is expected to be run whenever an icon is added or removed to an icon theme (typically an application icon into `hicolor` theme) and some programs do indeed run this after icon installation. However, since packages are installed into their own prefix by Nix, this would lead to conflicts. For that reason, `gtk3` provides a [setup hook](#ssec-gnome-hooks-gtk-drop-icon-theme-cache) that will clean the file from installation. Since most applications only ship their own icon that will be loaded on start-up, it should not affect them too much. On the other hand, icon themes are much larger and more widely used so we need to cache them. Because we recommend installing icon themes globally, we will generate the cache files from all packages in a profile using a NixOS module. You can enable the cache generation using `gtk.iconCache.enable` option if your desktop environment does not already do that.
 
+It's also worth noting that some GTK4 packages rely on `gtk-update-icon-cache` to generate icon theme cache in Meson build scripts. They can be replaced with `gtk4-update-icon-cache` in GTK4 package, without adding GTK3 to `nativeBuildInputs`. This is usually something like:
```

Comment:
I am not fond of documenting downstream hacks for stuff that should be fixed upstream. Opened a tracking issue https://github.com/NixOS/nixpkgs/issues/302605

Reply:
Okay. Dropping this commit as of now.